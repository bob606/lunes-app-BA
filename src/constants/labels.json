{
  "overview": {
    "welcome": "Willkommen bei Lunes!\nLerne Vokabeln für deinen Beruf.",
    "unit": "Lektion",
    "units": "Lektionen",
    "word": "Wort",
    "words": "Wörter",
    "exercises": "Übungen"
  },
  "header": {
    "overview": "Übersicht",
    "overviewExercises": "Übersicht der Übungen",
    "cancelExercise": "Übung beenden"
  },
  "exercises": {
    "overview": {
      "title": "Vokabelübersicht",
      "description": "Alle Wörte anschauen"
    },
    "wordSingleChoice": {
      "title": "Wort auswählen",
      "description": "Wähle das richtige Wort"
    },
    "articleSingleChoice": {
      "title": "Artikel üben",
      "description": "Wähle den richtigen Artikel"
    },
    "write": {
      "title": "Wort schreiben",
      "description": "Wort mit Artikel eingeben",
      "placeholder": {
        "insertAnswer": "Wort mit Artikel eingeben",
        "newTry": "Neuer Versuch"
      },
      "button": {
        "checkInput": "Eingabe überprüfen",
        "showSolution": "Lösung anzeigen",
        "tryLater": "Später versuchen"
      },
      "feedback": {
        "articleMissing": "Ups, deine Eingabe ist nicht vollständig. Füge einen Artikel ein.",
        "wrong": "Die richtige Antwort ist: ",
        "almostCorrect1": "Deine Eingabe ",
        "almostCorrect2": " ist fast richtig.",
        "correct": "Toll, weiter so! \nDeine Eingabe ist richtig."
      }
    },
    "buttons": {
      "next": "Nächstes Wort",
      "showResults": "Ergebnisse anzeigen"
    }
  },
  "results": {
    "resultsOverview": "Ergebnis-Übersicht",
    "feedbackGood": "Toll, weiter so! \nDu hast die Übung sehr gut gemeistert.",
    "feedbackMedium": "Übung macht den Meister, \nversuche es noch einmal!",
    "feedbackBad": "Nicht aufgeben! \nVersuche es noch einmal!",
    "checkEntries": "Eingaben überprüfen",
    "retryExercise": "Übung wiederholen",
    "wrong": "falsche",
    "similar": "ähnliche",
    "correct": "richtige",
    "entries": "Eingaben",
    "viewEntries": "Eingaben anschauen",
    "show": "Zeige",
    "of": "von"
  },
  "cancelModal": {
    "cancelAsk": "Möchtest du diese Übung wirklich beenden?",
    "cancel": "Beenden",
    "continue": "Weiter machen"
  }
}
