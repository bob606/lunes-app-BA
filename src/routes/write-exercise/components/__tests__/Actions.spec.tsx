import { render } from '@testing-library/react-native'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import React from 'react'
import 'react-native'

import labels from '../../../../constants/labels.json'
import wrapWithTheme from '../../../../testing/wrapWithTheme'
import Action, { IActionsProps } from '../Actions'

describe('Components', () => {
  describe('Actions', () => {
    const defaultActionProps: IActionsProps = {
      input: '',
      result: null,
      isFinished: false,
      giveUp: () => {},
      tryLater: () => {},
      checkEntry: () => {},
      continueExercise: () => {},
      needsToBeRepeated: false
    }

    it('renders correctly across screens', () => {
      const component = shallow(<Action {...defaultActionProps} />)
      expect(toJson(component)).toMatchSnapshot()
    })

    it('should render "check out" button when there is a result and isFinished is true', () => {
      const actionProps: IActionsProps = {
        ...defaultActionProps,
        result: 'correct',
        isFinished: true
      }

      const { getByText } = render(<Action {...actionProps} />, { wrapper: wrapWithTheme })
      expect(getByText(labels.exercises.showResults)).toBeDefined()
    })

    it('should render "next word" button when there is a result and isFinished is false', () => {
      const actionProps: IActionsProps = {
        ...defaultActionProps,
        result: 'correct',
        isFinished: false
      }

      const { getByText } = render(<Action {...actionProps} />, { wrapper: wrapWithTheme })
      expect(getByText(labels.exercises.next)).toBeDefined()
    })

    it('should render "check entry", "i give up", "try later" buttons when there is no result and isFinished is false', () => {
      const actionProps: IActionsProps = {
        ...defaultActionProps,
        isFinished: false
      }

      const { getByText } = render(<Action {...actionProps} />, { wrapper: wrapWithTheme })
      expect(getByText(labels.exercises.write.checkInput)).toBeDefined()
      expect(getByText(labels.exercises.tryLater)).toBeDefined()
      expect(getByText(labels.exercises.write.showSolution)).toBeDefined()
    })

    it('should render "check entry", "i give up" buttons when there is no result and isFinished is true', () => {
      const actionProps: IActionsProps = {
        ...defaultActionProps,
        isFinished: true
      }

      const { getByText } = render(<Action {...actionProps} />, { wrapper: wrapWithTheme })
      expect(getByText(labels.exercises.write.checkInput)).toBeDefined()
      expect(getByText(labels.exercises.write.showSolution)).toBeDefined()
    })

    it('should render disabled "check entry" button when input is empty', () => {
      const actionProps: IActionsProps = {
        ...defaultActionProps
      }

      const { getByText } = render(<Action {...actionProps} />, { wrapper: wrapWithTheme })
      expect(getByText(labels.exercises.write.checkInput)).toBeDisabled()
    })
  })
})
