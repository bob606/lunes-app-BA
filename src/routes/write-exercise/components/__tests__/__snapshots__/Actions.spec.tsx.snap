// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Actions renders correctly across screens 1`] = `
<Fragment>
  <Button
    buttonTheme="dark"
    disabled={true}
    onPress={[Function]}
    testID="check-entry"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#f9fafb",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 30,
            "fontWeight": "600",
            "letterSpacing": 0.4,
            "textAlign": "center",
            "textTransform": "uppercase",
          },
          Object {
            "color": "#777e91",
          },
        ]
      }
    >
      Eingabe überprüfen
    </Text>
  </Button>
  <Button
    buttonTheme="light"
    onPress={[Function]}
    testID="give-up"
  >
    <Text
      style={
        Object {
          "color": "#000e38",
          "fontFamily": "SourceSansPro-SemiBold",
          "fontSize": 30,
          "fontWeight": "600",
          "letterSpacing": 0.4,
          "textAlign": "center",
          "textTransform": "uppercase",
        }
      }
    >
      Lösung anzeigen
    </Text>
  </Button>
  <Button
    onPress={[Function]}
    testID="try-later"
  >
    <Text
      style={
        Object {
          "color": "#000e38",
          "fontFamily": "SourceSansPro-SemiBold",
          "fontSize": 30,
          "fontWeight": "600",
          "letterSpacing": 0.4,
          "textAlign": "center",
          "textTransform": "uppercase",
        }
      }
    >
      Später versuchen
    </Text>
    <SvgrURL
      style={
        Object {
          "marginLeft": 5,
        }
      }
    />
  </Button>
</Fragment>
`;
